libnetwork/ipams/defaultipam/allocator_test.go

```diff
@@ -25,34 +25,52 @@ import (
 
 func TestKeyString(t *testing.T) {
 	k := &PoolID{AddressSpace: "default", SubnetKey: SubnetKey{Subnet: netip.MustParsePrefix("172.27.0.0/16")}}
-	expected := "default/172.27.0.0/16"
-	if expected != k.String() {
-		t.Fatalf("Unexpected key string: %s", k.String())
-	}
+	expected := "PoolID{AddressSpace=default;Subnet=172.27.0.0/16}"
+	assert.Equal(t, k.String(), expected)
 
 	k2, err := PoolIDFromString(expected)
-	if err != nil {
-		t.Fatal(err)
-	}
+	assert.NilError(t, err)
+
 	if k2.AddressSpace != k.AddressSpace || k2.Subnet != k.Subnet {
-		t.Fatalf("SubnetKey.FromString() failed. Expected %v. Got %v", k, k2)
+		t.Fatalf("PoolIDFromString() failed. Expected %v. Got %v", k, k2)
 	}
 
-	expected = fmt.Sprintf("%s/%s", expected, "172.27.3.0/24")
 	k.ChildSubnet = netip.MustParsePrefix("172.27.3.0/24")
-	if expected != k.String() {
-		t.Fatalf("Unexpected key string: %s", k.String())
-	}
+	expected = "PoolID{AddressSpace=default;Subnet=172.27.0.0/16;ChildSubnet=172.27.3.0/24}"
+	assert.Equal(t, k.String(), expected)
 
 	k2, err = PoolIDFromString(expected)
-	if err != nil {
-		t.Fatal(err)
-	}
+	assert.NilError(t, err)
+
 	if k2.AddressSpace != k.AddressSpace || k2.Subnet != k.Subnet || k2.ChildSubnet != k.ChildSubnet {
 		t.Fatalf("SubnetKey.FromString() failed. Expected %v. Got %v", k, k2)
 	}
```

Comment:
Maybe worth testing PoolIDFromString with a v2 string that has no ChildSubnet? (And some invalid strings to check the InvalidParameter code paths?)

(Could also convert more of the old test to use Equal rather than comparing individual fields, and use `assert.Check(t, is.Equal(...))`.)